File: frontend/src/types/enrollment.ts
export interface Enrollment {
  id: number;
  courseId: number;
  learnerId: number;
  courseTitle: string;
  learnerName: string;
  status: string;
  enrolledOn: string;
}

export interface EnrollmentCreateRequest {
  courseId: number;
  learnerId: number;
}

export interface CourseEnrollment {
  id: number;
  learnerId: number;
  learnerName: string;
  learnerEmail: string;
  courseId: number;
  status: string;
  enrolledOn: string;
}

File: backend/TrainingManagement.Api/Models/Enrollment.cs
using System.ComponentModel.DataAnnotations;
using TrainingManagement.Api.Enums;

namespace TrainingManagement.Api.Models;

public class Enrollment
{
    public int Id { get; set; }

    public int CourseId { get; set; }

    public int LearnerId { get; set; }

    public DateTime EnrolledOn { get; set; }

    public StatusEnum Status { get; set; }

    public Course Course { get; set; } = null!;

    public Learner Learner { get; set; } = null!;
}
